src/node/desktop/test/unit/main/desktop-options.test.ts

```diff
@@ -127,4 +134,125 @@ describe('DesktopOptions', () => {
     const options2 = testingDesktopOptions();
     assert.equal(options2.zoomLevel(), newZoom);
   });
+  it('restores window bounds to correct display', () => {
+    const displays = [{workArea: {width: 2000, height: 2000, x: 0, y: 0}}, {workArea: {width: 2000, height: 2000, x: 2000, y: 0}}];
+    const savedWinBounds = {width: 500, height: 500, x: 2100, y: 100};
+
+    // Save bounds onto a secondary display on the right
+    DesktopOptions().saveWindowBounds(savedWinBounds);
+
+    const sandbox = sinon.createSandbox();
+    sandbox.stub(screen, 'getAllDisplays').returns(displays as Display[]);
+    const testMainWindow = sandbox.createStubInstance(BrowserWindow);
+    testMainWindow.setBounds.withArgs(savedWinBounds);
+    testMainWindow.getSize.returns([savedWinBounds.width, savedWinBounds.height]);
+    
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Kinda annoying to have these in unit testing to keep the linter happy. I couldn't figure out how to get it to compile without adding `as any`

Reply:
I'll take a peek once it is merged, should rarely need to use 'any' or do a cast like this (though sometimes it probably will be necessary, and this might be one of those times).

Reply:
Fixed via #9651